src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax;
+
+public class Main {
+
+    public static void main(String[] Args) {
+        final int ARRAY_LENGHT = 6;
+        Figure figures[] = new Figure[6];
+
+        for (int i = 0; i < ARRAY_LENGHT; i++) {
+            if (i < ARRAY_LENGHT / 2) {
+                figures[i] = new FigureSupplier().getRandomFigure();
+                figures[i].draw();
+            } else {
+                figures[i] = new FigureSupplier().getDefaultFigure();
+                figures[i].draw();
```

Comment:
Creating a new `FigureSupplier` instance in each iteration is not efficient. Instantiate `FigureSupplier` once before the loop and use that instance within the loop.